Mathlib/AlgebraicTopology/SimplicialSet/Coskeletal.lean

```diff
@@ -0,0 +1,270 @@
+      exact congr_fun (s.w α).symm x
+    · dsimp only [spineEquiv, Equiv.coe_fn_mk]
+      dsimp
+      rw [← FunctorToTypes.map_comp_apply]
+      let α : strArrowMk₂ f hi ⟶ strArrowMk₂ (mkOfSucc k ≫ f) (by omega) :=
+        StructuredArrow.homMk (mkOfSucc k).op (by simp; rfl)
+      exact (isPointwiseRightKanExtensionAt.fac_aux₃ _ _ _ _).trans (congr_fun (s.w α).symm x)
+  uniq s m hm := by
+    ext x
+    apply StrictSegal.spineInjective (X := X)
+    dsimp [spineEquiv]
+    rw [StrictSegal.spine_spineToSimplex]
+    ext i
+    · exact congr_fun (hm (StructuredArrow.mk (Y := op [0]₂) ([0].const [n] i).op)) x
+    · exact congr_fun (hm (.mk (Y := op [1]₂) (.op (mkOfLe _ _ (Fin.castSucc_le_succ i))))) x
+
+/-- Since `rightExtensionInclusion₂IsPointwiseRightKanExtensionAt X n` proves that the appropriate
+cones are limit cones, `rightExtensionInclusion X 2` is a pointwise right Kan extension.-/
+noncomputable def isPointwiseRightKanExtension :
+    RightExtension.IsPointwiseRightKanExtension (rightExtensionInclusion X 2) :=
+  fun Δ => isPointwiseRightKanExtensionAt X Δ.unop.len
+
+theorem isRightKanExtension :
+    (rightExtensionInclusion X 2).left.IsRightKanExtension (rightExtensionInclusion X 2).hom :=
+  RightExtension.IsPointwiseRightKanExtension.isRightKanExtension
+    (isPointwiseRightKanExtension X)
+
+/-- When `X` is `StrictSegal`, `X` is 2-coskeletal. -/
+instance isCoskeletal : SimplicialObject.IsCoskeletal X 2 where
+  isRightKanExtension := isRightKanExtension X
+
+/-- The canonical isomorphism `X ≅ (cosk 2).obj X` defined when `X` is `StrictSegal`. -/
+noncomputable def isoCosk2 : X ≅ (cosk 2).obj X := SimplicialObject.isoCoskOfIsCoskeletal X 2
+
+end StrictSegal
+
+end
+
+end SSet
+
+namespace CategoryTheory
+
+namespace Nerve
+
+open SSet
+
+instance isCoskeletal (C : Type u) [Category.{v} C] :
+    SimplicialObject.IsCoskeletal (nerve C) 2 := by infer_instance
+
+/-- The essential data of the nerve functor is contained in the 2-truncation, which is
+recorded by the composite functor `nerveFunctor₂`.-/
+def nerveFunctor₂ : Cat.{v, u} ⥤ SSet.Truncated 2 := nerveFunctor ⋙ truncation 2
+
+/-- The counit of `coskAdj 2` defines a natural transformation from the nerve to the right
+Kan extension of the 2-truncated nerve.-/
+noncomputable def cosk2NatTrans : nerveFunctor.{u, v} ⟶ nerveFunctor₂ ⋙ Truncated.cosk 2 :=
+  whiskerLeft nerveFunctor (coskAdj 2).unit
+
+/-- The natural isomorphism between `nerveFunctor` and `nerveFunctor ⋙ Truncated.cosk 2` whose
+components `nerve C ≅  (Truncated.cosk 2).obj (nerve₂ C)` shows that nerves of categories are
+2-coskeletal.-/
+noncomputable def cosk2Iso : nerveFunctor.{u, u} ≅ nerveFunctor₂.{u, u} ⋙ Truncated.cosk 2 := by
+  refine NatIso.ofComponents ?_ ?_
+  · intro C
+    dsimp [nerveFunctor, nerveFunctor₂]
+    exact (StrictSegal.isoCosk2 (nerve C))
+  · simp only [nerveFunctor_obj, comp_obj, id_eq, Functor.comp_map]
+    convert cosk2NatTrans.naturality
```

Comment:
```suggestion
/-- The natural isomorphism between `nerveFunctor` and `nerveFunctor ⋙ Truncated.cosk 2` whose
components `nerve C ≅ (Truncated.cosk 2).obj (nerve₂ C)` shows that nerves of categories are
2-coskeletal.-/
noncomputable def cosk₂Iso : nerveFunctor.{u, v} ≅ nerveFunctor₂.{v, u} ⋙ Truncated.cosk 2 :=
  NatIso.ofComponents (fun C ↦ (nerve C).isoCoskOfIsCoskeletal 2)
    (fun _ ↦ (coskAdj 2).unit.naturality _)
```

Then, you may remove the definition `cosk2NatTrans`, because it is always an isomorphism.
You may also fix the definition I had made of `nerveFunctor` by making the universe explicit:
```lean
def nerveFunctor : Cat.{v, u} ⥤ SSet where
```
Then, this will fix the order of the universes, and in this file `Coskeletal.lean`, the definition would become:
```lean
noncomputable def cosk₂Iso : nerveFunctor.{v, u} ≅ nerveFunctor₂.{v, u} ⋙ Truncated.cosk 2 :=
```
(As this may be confusing, please let me know if this does not fully make sense to you.)

Reply:
Done.